### PR TITLE
Add tests for string and f-string literals

### PR DESCRIPTION
**Pull request by dhruvmanila**
## Summary

This PR provides additional test cases for string and f-string parsing.

There are a few things the parser does for strings which isn't really ideal for error recovery. This PR adds test cases to showcase them (along with TODO comments) which will be helpful when working on error recovery. This requires some changes on the lexer side especially w.r.t. unterminated strings.

### CODE REVIEW

**Comment by dhruvmanila** [comment]
(I need to rebase the parser stack for the ecosystem check to be in sync, I'll do it later.)

File: crates/ruff_python_parser/src/parser/expression.rs [hunk 1]
```diff
@@ -1225,6 +1243,28 @@ impl<'src> Parser<'src> {
             }
         }
```
Comment (MichaReiser) [inline crates/ruff_python_parser/src/parser/expression.rs:1245]: Github doesn't allow me to comment on the relevant line: 

```rust
                    for string in strings {
                        values.push(match string {
                            StringType::Bytes(value) => value,
                            _ => ast::BytesLiteral::invalid(string.range()),
                        });
                    }
```

For the above code, isn't it impossible to have an element that isn't `StringType::Bytes`, given that `bytes_literal_count == strings.len()`?

File: crates/ruff_python_parser/src/parser/expression.rs [hunk 2]
```diff
@@ -1198,6 +1208,14 @@ impl<'src> Parser<'src> {
...
             match byte_literal_count.cmp(&strings.len()) {
                 Ordering::Less => {
+                    // TODO(dhruvmanila): This is not an ideal recovery because the parser
+                    // replaces the byte literals with an invalid string literal node. Any
+                    // downstream tools can extract the raw bytes from the range.
+
+                    // test_err mixed_bytes_and_non_bytes_literals
+                    // 'first' b'second'
+                    // f'first' b'second'
+                    // 'first' f'second' b'third'
```
Comment (MichaReiser) [inline crates/ruff_python_parser/src/parser/expression.rs:1218]: Yeah. We probably want to convert the node to a string and mark it as "errornous". We might also need to be slightly more clever by either marking the string or byte literals as errors, depending on which nodes are fewer. But that's something we can do later.

**Review by MichaReiser — APPROVED** [review]

File: crates/ruff_python_parser/src/parser/expression.rs (hunk 2 above)
Comment (dhruvmanila) [reply crates/ruff_python_parser/src/parser/expression.rs:1218]: Yeah, I had a similar idea. I'll add this to the TODO comment.

**Pull request merged**